sql-plugin/src/main/scala/com/nvidia/spark/rapids/aggregate.scala

```diff
@@ -469,6 +545,145 @@ object GpuHashAggregateIterator extends Logging {
   }
 }
 
+object GpuAggFirstPassIterator {
+  def apply(cbIter: Iterator[ColumnarBatch],
+      aggHelper: AggHelper,
+      metrics: GpuHashAggregateMetrics): Iterator[SpillableColumnarBatch] = {
+    val preprocessProjectIter = cbIter.map { cb =>
+      val sb = SpillableColumnarBatch (cb, SpillPriorities.ACTIVE_ON_DECK_PRIORITY)
+      aggHelper.preStepBound.projectAndCloseWithRetrySingleBatch (sb)
+    }
+    computeAggregateWithoutPreprocessAndClose(metrics, preprocessProjectIter, aggHelper)
+  }
+}
+
+// Partial mode:
+//  1. boundInputReferences: picks column from raw input
```

Comment:
not related to your change, `boundInputReferences` is no longer here :( we should remove from the comments (I missed this)